File: cslavb/Templates/DynamicRootList.vb
Imports System.Data.SqlClient

<Serializable()> _
Public Class DynamicRootList
  Inherits EditableRootListBase(Of DynamicRoot)

#Region " Business Methods "

  Protected Overrides Function AddNewCore() As Object

    Dim item As DynamicRoot = DynamicRoot.NewDynamicRoot
    Add(item)
    Return item

  End Function
#End Region

#Region " Authorization Rules "

  Public Shared Function CanGetObject() As Boolean
    ' TODO: customize to check user role
    Return ApplicationContext.User.IsInRole("")
  End Function

  Public Shared Function CanEditObject() As Boolean
    ' TODO: customize to check user role
    Return ApplicationContext.User.IsInRole("")
  End Function

#End Region

#Region " Factory Methods "

  Public Shared Function NewDynamicRootList() As DynamicRootList
    Return New DynamicRootList()
  End Function

  Public Shared Function GetDynamicRootList() As DynamicRootList
    Return DataPortal.Fetch(Of DynamicRootList)()
  End Function

  Private Sub New()
    ' require use of factory methods
    AllowNew = True
  End Sub

#End Region

#Region " Data Access "

  Private Overloads Sub DataPortal_Fetch()

    ' TODO: load values
    RaiseListChangedEvents = False
    Using dr As SqlDataReader = Nothing
      While dr.Read
        Add(DynamicRoot.GetDynamicRoot(dr))
      End While
    End Using
    RaiseListChangedEvents = True

  End Sub

#End Region

End Class

File: cslacs/Templates/DynamicRootList.cs
using System;
using System.Data.SqlClient;
using Csla;

[Serializable()]
public class DynamicRootList : 
  EditableRootListBase<DynamicRoot>
{

  #region Business Methods

  protected override object AddNewCore()
  {
    DynamicRoot item = DynamicRoot.NewDynamicRoot();
    Add(item);
    return item;
  }

  #endregion

  #region  Authorization Rules

  public static bool CanGetObject()
  {
    // TODO: customize to check user role
    return ApplicationContext.User.IsInRole("");
  }

  public static bool CanEditObject()
  {
    // TODO: customize to check user role
    return ApplicationContext.User.IsInRole("");
  }

#endregion

#region  Factory Methods 

  public static DynamicRootList NewDynamicRootList()
  {
    return new DynamicRootList();
  }

  public static DynamicRootList GetDynamicRootList()
  {
    return DataPortal.Fetch<DynamicRootList>();
  }

  private DynamicRootList()
  {
    // require use of factory methods
    AllowNew = true;
  }

#endregion

#region  Data Access 

  private void DataPortal_Fetch()
  {

    // TODO: load values
    RaiseListChangedEvents = false;
    using (SqlDataReader dr = null)
    {
      while (dr.Read())
      {
        Add(DynamicRoot.GetDynamicRoot(dr));
      }
    }
    RaiseListChangedEvents = true;

  }

#endregion

}
